Mathlib/Order/Normal.lean

```diff
@@ -0,0 +1,156 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Order.SuccPred.InitialSeg
+
+/-!
+# Normal functions
+
+A normal function between well-orders is a strictly monotonic continuous function. Normal functions
+arise chiefly in the context of cardinal and ordinal-valued functions. Unfortunately, Mathlib places
+these earlier in the import chain than the topological notion of continuity.
```

Comment:
I find it bad to motivate such changes by import concerns. At least, we have to be very sure there's no way to tame the imports, eg here to reduce the imports to continuity until it's okay to import it in the theory of ordinal-valued functions.

Have you tried that?

Reply:
There was a brief experiment to make ordinals import topology, and it added some hundred or so imports.

Some basic ordinal arithmetic is required to set up cardinal arithmetic, which are then needed in linear algebra, model theory, or even to some extent to characterize `Nat.card`. Adding even basic topology as an import to all of these things would be pretty catastrophic.

The characterization of ordinal functions as continuous and strictly monotone functions isn't even all too useful within ordinal arithmetic. For the most part, one really only cares about the fact that normal functions "commute" with suprema, i.e. `f (sSup s) = sSup (f '' s)`, which is what this definition takes as its characterization.

Reply:
None of what you said seems to precisely contradict my point that you could reduce imports to continuity.

Can you at least bring it up on Zulip?

Reply:
> The characterization of ordinal functions as continuous and strictly monotone functions isn't even all too useful within ordinal arithmetic.

Can you at least make this the motivation rather than the import reduction?

Reply:
> The characterization of ordinal functions as continuous and strictly monotone functions isn't even all too useful within ordinal arithmetic. For the most part, one really only cares about the fact that normal functions "commute" with suprema, i.e. `f (sSup s) = sSup (f '' s)`, which is what this definition takes as its characterization.

I'd say this is a better motivation, and should be mentioned in the module docstring, as opposed to import concerns.